File: soal1.ts
function countingValleys(steps: number, path: string): number {
    let seaLevel = 0;
    let valleys = 0;
    let currentLevel = 0;

    for (let i = 0; i < steps; i++) {
        if (path[i] === 'U') {
            currentLevel++;
            if (currentLevel === 0) {
                valleys++;
            }
        } else {
            currentLevel--;
        }
    }

    return valleys;
}

// Contoh penggunaan:
const steps = 8;
const path = "DDUUUUDD";
console.log(countingValleys(steps, path)); // Output: 1

File: soal1.py
def countingValleys(steps, path):
    sea_level = 0
    valleys = 0
    current_level = 0

    for step in path:
        if step == 'U':
            current_level += 1
        else:
            current_level -= 1

        # Jika naik ke level 0 dari bawah, berarti baru saja keluar dari lembah
        if current_level == 0 and step == 'U':
            valleys += 1

    return valleys

# Contoh penggunaan
if __name__ == "__main__":
    steps = int(input("Masukkan jumlah langkah: "))
    path = input("Masukkan jalur pendakian: ").strip()
    print(countingValleys(steps, path))
